File: src/test/java/org/micoli/php/builders/CodeStyleBuilder.java
package org.micoli.php.builders;

import org.micoli.php.codeStyle.configuration.CodeStyle;

public class CodeStyleBuilder {
    private final CodeStyle codeStyle;

    private CodeStyleBuilder() {
        this.codeStyle = new CodeStyle();
    }

    public static CodeStyleBuilder create() {
        return new CodeStyleBuilder();
    }

    public CodeStyleBuilder withStyleAttribute(String attribute) {
        codeStyle.styleAttribute = attribute;
        return this;
    }

    public CodeStyleBuilder withValue(String value) {
        codeStyle.value = value;
        return this;
    }

    public CodeStyle build() {
        return codeStyle;
    }
}

File: src/test/kotlin/org/micoli/php/builders/CodeStyleBuilder.kt
package org.micoli.php.builders

import org.micoli.php.codeStyle.configuration.CodeStyle

class CodeStyleBuilder private constructor() {
    private val codeStyle: CodeStyle = CodeStyle()

    fun withStyleAttribute(attribute: String): CodeStyleBuilder {
        codeStyle.styleAttribute = attribute
        return this
    }

    fun withValue(value: String): CodeStyleBuilder {
        codeStyle.value = value
        return this
    }

    fun build(): CodeStyle {
        return codeStyle
    }

    companion object {
        @JvmStatic
        fun create(): CodeStyleBuilder {
            return CodeStyleBuilder()
        }
    }
}
